types/block.go

```diff
@@ -1148,6 +1148,9 @@ type Data struct {
 	// This means that block.AppHash does not include these txs.
 	Txs Txs `json:"txs"`
 
+	// MetaData is a arbitrary byte slice. This field will vary on the use case of the application
+	// NOTE: can be empty or populated.
+	MetaData []byte `json:"meta_data"`
```

Comment:
This makes sense for the general case. I think here, we can either populate the `Messages` and later the `IntermediateStateRoots` fields directly (closer to the LL spec), or, assume that the returned `MetaData` is all namespaced blobs (aka Messages) including those with a reserved namespace ID that are not Tx (closer to tendermint when this gets upstreamed). 
I think, for now, we should rather try to be closer to the LL spec here and remove the general MetaData field and use the concrete LazyLedger-specific fields.